### PR TITLE
refactor(dockerfile): refactor dockerfile

### PR DESCRIPTION
**Pull request by mritd**
refactor dockerfile

Signed-off-by: mritd <mritd@linux.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
+    # Build and install
+    && cmake . && make -j $(nproc) && strip -s trojan \
+    && mv trojan /usr/local/bin) \
+    # Add runtime dependencies
+    && apk add --virtual .trojan-rundeps \
         libstdc++ \
         boost-system \
         boost-program_options \
-        mariadb-connector-c
-WORKDIR /config
```
Comment (GreaterFire) [inline Dockerfile]: Actually, I kinda want the user to mount the whole `/config` directory, which may contain the private key file and the certificate. By setting `WORKDIR` to `/config`, the user can just set the relative path of these files in the config file.

Comment (mritd) [reply Dockerfile]: This is a good idea 😁

File: Dockerfile [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
-    && cmake . \
-    && make \
-    && strip -s trojan \
-    && mv trojan .. \
-    && cd .. \
-    && rm -rf repo \
-    && apk del .build-deps \
-    && apk add --no-cache --virtual .trojan-rundeps \
+    # Get trojan source code and checkout to the specified branch
+    && git clone https://github.com/trojan-gfw/trojan.git \
```
Comment (GreaterFire) [inline Dockerfile]: What about `git clone --branch=${VERSION}`?

Comment (mritd) [reply Dockerfile]: ... I just think that the branch option description is not clear, in fact we don't have the v1.12.2 branch (it's just a tag), so I'm executing the checkout command 😂

To optimize the Dockerfile, we can use the `--branch` option.

**Pull request merged**